src/node/internal/internal_http_incoming.ts

```diff
@@ -101,20 +102,30 @@ export class IncomingMessage extends Readable implements _IncomingMessage {
   }
 
   async #tryRead(): Promise<void> {
-    if (this._stream == null) return;
+    if (this._stream == null || this.#reading) return;
+
+    this.#reading = true;
 
     try {
       this.#reader ??= this._stream.getReader();
-      const data = await this.#reader.read();
-      if (data.done) {
-        // Done with stream, tell Readable we have no more data;
-        this.complete = true;
-        this.push(null);
-      } else {
-        this.push(data.value);
+
+      while (!this.destroyed) {
```

Comment:
This ends up reading the stream to completion even when we just call `read()`. Is that actually what we want? The definition of `read()` is that it gets the next chunk of data. This is certainly the correct behavior for `pipe()` but is it the right thing for `read()`?

Reply:
Ah, nevermind I see the `this.push` backpressure signal checking below 👍🏻